lib/app/logger.dart

```diff
@@ -0,0 +1,135 @@
+import 'dart:convert';
+
+import 'package:logger/logger.dart';
+// ignore: implementation_imports
+import 'package:logger/src/ansi_color.dart';
+import 'package:meta/meta.dart';
+import 'package:time_machine/time_machine.dart';
+import 'package:time_machine/time_machine_text_patterns.dart';
+
+final logger = Logger(printer: FancyPrinter());
+
+/// A [LogPrinter] similar to [PrettyPrinter], but with less vertical space.
+class FancyPrinter extends LogPrinter {
+  static final levelEmojis = {
+    Level.verbose: '💤',
+    Level.debug: '🐛',
+    Level.info: '💡',
+    Level.warning: '⚠️',
+    Level.error: '⛔',
+    Level.wtf: '👾',
+  };
+
+  static final timeFormat =
+      LocalTimePattern.createWithInvariantCulture('HH:mm:ss;fff');
+  static const stackTraceMethodCount = 10;
+
+  static const verticalSpace = '   ';
+  static const verticalLine = ' │ ';
+  static const middleCorner = ' ├─';
+  static const bottomCorner = ' └─';
+
+  @override
+  void log(LogEvent event) {
+    final messageStr = stringifyMessage(event.message) ?? '';
+    final timeStr = timeFormat.format(Instant.now().inLocalZone().clockTime);
+    String stackTraceStr;
+    if (event.stackTrace != null ||
+        [Level.error, Level.wtf, Level.warning].contains(event.level)) {
```

Comment:
We should probably list these error levels in order of their severity:
```suggestion
        [Level.warning, Level.error, Level.wtf].contains(event.level)) {
```